src/components/CheckboxInput/CheckboxInput.tsx

```diff
@@ -31,32 +35,24 @@ export type CheckboxInputProps = CheckboxHTMLElementProps & {
 export const CheckboxInput = React.forwardRef<
   HTMLInputElement,
   CheckboxInputProps
->(({ checked, className, disabled, ...other }, ref) => {
-  // Make indeterminate checkbox visually match the colors of a checked state, but announce itself
-  // as "mixed" to screen readers
-  //
-  // However, https://html.spec.whatwg.org/multipage/input.html#checkbox-state-(type=checkbox)
-  // seems to consider `indeterminate` as a separate bit of information than `checked`. They can be
-  // in all 4 combinations (only 3 visual states, though).
-  //
-  // Should we make them separate props?
-  const checkedProps =
-    checked === 'indeterminate'
-      ? {
-          'aria-checked': 'mixed' as const,
```

Comment:
Previously we set `aria-checked` so screen readers knew about it, but the DOM itself didn't consider the checkbox to be indeterminate.

Here we fix that (which unfortunately requires JS and jumping through a hoop).